### ISSUE TITLE
Can I distribute the stages over multiple GPUs? Like you see below

### ISSUE THREAD

**Issue by phalexo** [deep-floyd/IF#50]
from deepfloyd_if.modules import IFStageI, IFStageII, StableStageIII
from deepfloyd_if.modules.t5 import T5Embedder

# I have 4 Maxwel Titan X with 12GB VRAM each.
if_I = IFStageI('IF-I-XL-v1.0', device='cuda:0')
if_II = IFStageII('IF-II-L-v1.0', device='cuda:1')
if_III = StableStageIII('stable-diffusion-x4-upscaler', device='cuda:2')
t5 = T5Embedder(device="cuda:3")

**Comment by brycedrennan** [comment]
did you try it?

**Comment by phalexo** [comment]
> 

No, to try it I have to provide my info to download the checkpoints. If the above is not possible with the current code, then I don't want to register for the checkpoints.

**Comment by ThioJoe** [comment]
Seems to work.

Before:
<img width="416" alt="image" src="https://user-images.githubusercontent.com/12518330/235280081-53fb22ff-f60d-4de9-9f5f-0ce4dd297b13.png">
<img width="164" alt="image" src="https://user-images.githubusercontent.com/12518330/235280072-750ce780-67d5-4981-819a-cc0c38f135e8.png">

After:
<img width="437" alt="image" src="https://user-images.githubusercontent.com/12518330/235280102-da912f00-0e0f-4db7-8abc-64b43542c145.png">
<img width="166" alt="image" src="https://user-images.githubusercontent.com/12518330/235280126-0f789c08-2a16-4c23-93b1-adb07b233664.png">

**Comment by phalexo** [comment]
> Seems to work.
> 
>
Can I assume, you were also able to get an image out of it? Since they are using "accelerate" library already, I hoped it would work.

Thanks for the info.

**Comment by ThioJoe** [comment]
Yes I can now confirm I was able to get images out of it with multi GPUs. 

Also you can change
`t5 = T5Embedder(device="cpu")` 
to be a GPU like
`t5 = T5Embedder(device="cuda:0")`

**Comment by phalexo** [comment]
> Yes I can now confirm I was able to get images out of it with multi GPUs.
> 
> Also you can change `t5 = T5Embedder(device="cpu")` to be a GPU like `t5 = T5Embedder(device="cuda:0")`

Did you do anything additional? I am running into an issue that cuBLAS is not supported when t5 is called. I have installed cuda toolset 11.8 after 11.3 and it still gives me the same error. 

Did the images display correctly?

Can you put up some info about your environment? Python module versions, kernel, etc...



Thanks.

**Comment by larme** [comment]
Hi all, we made an [integration of BentoML with IF](https://github.com/bentoml/IF-multi-GPUs-demo/). It support starting local server with a simple way to assign stages to arbitrary GPU available to the system. You can also build docker image and deploy it to production environment using [bentoctl](https://github.com/bentoml/bentoctl) or [BentoCloud](https://www.bentoml.com/bento-cloud/).
